### DISCUSSION TITLE
最新的关于减少cpu消耗的更改，大幅度的减少了缓存更新的频率。

### DISCUSSION THREAD

**Discussion by yyysuo** [pymumu/smartdns#1481] [Q&A]
https://github.com/pymumu/smartdns/commit/887ef7b20e416909ed31ee8b37dad23c7cf197d5

看了一下代码看不太懂，未激活的缓存是如何判断的呢？smartdns上游接了Adguard Home，缓存里面的域名大约有5万个，以前在Adguard Home的日志中，每天都能看到增加约5万个域名的请求，更新到最新代码之后，上游的更新请求几乎都没有了，看来这5万个域名，大部分都被判断为未激活状态的缓存了？那未激活状态的域名，缓存是一直不更新的还是只是更新的频率变低了？另外缓存更新的时间段能自己定义吗？

**Comment by pymumu** [comment]
开启过期缓存后，TTL超时的域名，会全部缓存到inactive缓存中。

inactive缓存中的数据，在开启了`prefetch-domain`的情况下，会使用`serve-expired-prefetch-time`设置的周期来更新。每2s更新25个域名。

当inactive中的数据，在`serve-expired-ttl`的时间后，没有任何访问时，将会被删除。

**Reply by yyysuo** [reply]
serve-expired-ttl 259200000
serve-expired-prefetch-time 86400
cache-checkpoint-time 36000

跟我理解的一样呀，跟之前的逻辑应该也没变化吧 。我设置的域名过期时间超级长，几年也不过期，5万域名，每24小时刷新一次缓存，上游的adgh就有大约5万访问记录 。用最新的源码，好像不再刷新缓存了，感觉好像是缓存更新功能出问题了 。

**Reply by pymumu** [reply]
应该是改了逻辑之后，对原缓存文件的处理有误，会导致部分域名的超时得不到及时处理。

还有，prefetch获取时间也和之前触发不一样，prefetch，之前prefetch时间是从域名第一次查询时+serve-expired-prefetch-time。
现在修改成了域名TTL过期后+serve-expired-prefetch-time。

你serve-expired-prefetch-time设置为1天的话，之前的版本会在1天只能触发查询。
现在的版本，会在1天+TTL的时间触发查询。

这个有一点差异。

可以等1~2天看看。

**Reply by yyysuo** [reply]
好的，更新最新源码应该有2天了，按理说应该有更新域名缓存的动作了 。再等一天，没更新的话我清空一下缓存试一下 。再不行回滚到官方release试试 。

**Reply by yyysuo** [reply]
按理说ttl是按分钟计的吧，是不是很少有按天计的，区别应该不大 。
